### ISSUE TITLE
Ability of dendroids works incorrect

### ISSUE THREAD

**Issue by GrafDeVan** [vcmi/vcmi#4349]
I binded champions, but they went ahead. Look video https://youtu.be/1czcEAJAILs

It's AIL build 1.6.0.df1dae22baf1e9b29be54a50f11abe5a18e77e22

save [чемпы.zip](https://github.com/user-attachments/files/16457188/default.zip) Dessa attack

**Comment by GeorgeK1ng** [comment]
Video not available, does same issue exist in  v1.6 latest daily build?

**Comment by Povelitelh3** [comment]
> Video not available, does same issue exist in v1.6 latest daily build?

ability dendroids doesn't work as intended. 1.6.4 beta (meet today)
the problem is that I myself don’t remember how to make the binding disappear...

**Comment by Povelitelh3** [comment]
@GeorgeK1ng I have found how reproduce this problem
video [dendr2.zip](https://github.com/user-attachments/files/18420246/dendr2.zip)
map [dendr.zip](https://github.com/user-attachments/files/18420257/dendr.zip)

**Comment by GrafDeVan** [comment]
Still related to 1.8 dev

**Comment by kambala-decapitator** [comment]
> the problem is that I myself don’t remember how to make the binding disappear

read docs? :) Binding is removed when Dendroid moves or dies.

from the video it seems that VCMI remembers only the first Dendroid that binds (as after first Dendroid moves and another one hits same target, the target can move), although when binding is re-applied, it should remember the last Dendroid who did it.

also visual debuff doesn't disappear from unit immediately when needed unlike oh3.

**Comment by GrafDeVan** [comment]
> read docs? :) Binding is removed when Dendroid moves or dies.
> 
> from the video it seems that VCMI remembers only the first Dendroid that binds (as after first Dendroid moves and another one hits same target, the target can move), although when binding is re-applied, it should remember the last Dendroid who did it.
> 
> also visual debuff doesn't disappear from unit immediately when needed unlike oh3.

Correct. With ability Androids something wrong, units can move even when binds.
suspect apparently, the bug is caused by the first stack that bound them moving away, and the game thinks the binding is no longer active. However, they are immediately attacked by two other stacks of Dendroids.

recorded today [Video_2026-06-25_192347.zip](https://github.com/user-attachments/files/29348573/Video_2026-06-25_192347.zip)
